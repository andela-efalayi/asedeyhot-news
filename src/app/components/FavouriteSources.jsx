import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardText } from 'material-ui/Card';

/**
 * Displays all favourite news sources on Favorites page
 * @function FavouriteSoures
 * @param {object} sources
 * @return {object} react-component
 */

const FavouriteSources = ({ sources }) => {
  if (sources === null) {
    return (
      <div>
        <h4>You have no saved sources.</h4>
      </div>
    );
  }
  return (
    <div>
      <Card className="col-xs-12">
        <CardText>
          <ul>
            {Object.entries(sources)
          .map(([key, value]) => (
            <li key={key}>{key}: <a href={value}>{value}</a></li>
          ))}
          </ul>
        </CardText>
      </Card>
    </div>
  );
};

FavouriteSources.propTypes = {
  sources: PropTypes.object
};

export default FavouriteSources;
