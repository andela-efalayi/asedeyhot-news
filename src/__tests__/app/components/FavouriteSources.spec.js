import React from 'react';
import { shallow, mount } from 'enzyme';
import FavouriteSources from '../../../app/components/FavouriteSources.jsx';

const emptyResponseFromFirebase = null;

const favouriteSources = {
  'Associated Press': 'http://bigstory.ap.org'
};

describe('FavouriteSources.jsx', () => {
  it('should match FavouriteSources component snapshot',
  () => {
    const wrapper = shallow(
        <FavouriteSources sources={favouriteSources}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should return "You have no saved sources" for a null response', () => {
    const wrapper = mount(
        <FavouriteSources sources={emptyResponseFromFirebase}/>
    );
    expect(wrapper.containsMatchingElement(
      <h4>You have no saved sources.</h4>)
    ).toEqual(true);
  });
});
