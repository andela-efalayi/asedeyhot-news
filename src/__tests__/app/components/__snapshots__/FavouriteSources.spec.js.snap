// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteSources.jsx should match FavouriteSources component snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Card
        className="col-xs-12"
        expandable={false}
        expanded={null}
        initiallyExpanded={false}
    >
        <CardText>
            <ul>
                <li>
                    Associated Press
                    : 
                    <a
                        href="http://bigstory.ap.org"
                    >
                        http://bigstory.ap.org
                    </a>
                </li>
            </ul>
        </CardText>
    </Card>
</div>,
  "nodes": Array [
    <div>
      <Card
            className="col-xs-12"
            expandable={false}
            expanded={null}
            initiallyExpanded={false}
      >
            <CardText>
                  <ul>
                        <li>
                              Associated Press
                              : 
                              <a
                                    href="http://bigstory.ap.org"
                              >
                                    http://bigstory.ap.org
                              </a>
                        </li>
                  </ul>
            </CardText>
      </Card>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <FavouriteSources
        sources={
                Object {
                        "Associated Press": "http://bigstory.ap.org",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "sources": Object {
            "Associated Press": "http://bigstory.ap.org",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Card
                    className="col-xs-12"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <CardText>
                              <ul>
                                        <li>
                                                  Associated Press
                                                  : 
                                                  <a
                                                            href="http://bigstory.ap.org"
                                                  >
                                                            http://bigstory.ap.org
                                                  </a>
                                        </li>
                              </ul>
                    </CardText>
          </Card>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Card
                    className="col-xs-12"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <CardText>
                              <ul>
                                        <li>
                                                  Associated Press
                                                  : 
                                                  <a
                                                            href="http://bigstory.ap.org"
                                                  >
                                                            http://bigstory.ap.org
                                                  </a>
                                        </li>
                              </ul>
                    </CardText>
          </Card>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FavouriteSources
    sources={
        Object {
            "Associated Press": "http://bigstory.ap.org",
          }
    }
/>,
}
`;
